R/expr__string.R

```diff
@@ -84,18 +84,103 @@ ExprStr_strptime = function(
 
     # Date
     datatype == pl$Date,
-    .pr$Expr$str_to_date(self, format, strict, exact, cache, ambiguous),
+    .pr$Expr$str_to_date(self, format, strict, exact, cache),
 
     # Time
     datatype == pl$Time,
-    .pr$Expr$str_to_time(self, format, strict, exact, cache, ambiguous),
+    .pr$Expr$str_to_time(self, format, strict, cache),
 
     # Other
     or_else = Err_plain("datatype should be of type {Date, Datetime, Time}")
   ) |>
     unwrap("in str$strptime():")
 }
 
+#' Convert a Utf8 column into a Date column
+#'
+#' @param format Format to use for conversion. See `?strptime` for possible
+#' values. Example: "%Y-%m-%d". If `NULL` (default), the format is
+#' inferred from the data. Notice that time zone `%Z` is not supported and will
+#' just ignore timezones. Numeric time zones like `%z` or `%:z`  are supported.
+#' @param strict If `TRUE` (default), raise an error if a single string cannot
+#' be parsed. If `FALSE`, parsing failure will produce a polars `null`.
+#' @param exact If `TRUE` (default), require an exact format match. Otherwise,
+#' allow the format to match anywhere in the target string.
+#' @param cache Use a cache of unique, converted dates to apply the datetime
+#' conversion.
+#'
+#' @return Expr
+#' @name ExprStr_to_date
+#'
+#' @examples
+#' pl$DataFrame(str_date = c("2009-01-02", "2009-01-03", "2009-1-4", "2009 05 01"))$
+#'   with_columns(date = pl$col("str_date")$str$to_date(strict = FALSE))
+ExprStr_to_date = function(format = NULL, strict = TRUE, exact = TRUE, cache = TRUE) {
+  .pr$Expr$str_to_date(self, format, strict, exact, cache) |>
+    unwrap("in $str$to_date():")
+}
```

Comment:
Probably applicable to other functions as well, but such functions may be better off using `...` to enforce named arguments.

Reply:
but then you lose the description of each param in the docs, no? In any case, if that concerns several other functions it can be done in another PR so I'm merging this

Reply:
> but then you lose the description of each param in the docs, no?

I'm sorry, but I don't understand. Here's what I mean:

```r
ExprStr_to_date = function(..., format = NULL, strict = TRUE, exact = TRUE, cache = TRUE)
```

Reply:
Oh right, could be added later but that could also apply to basically all functions that have several args